File: app/src/main/java/com/example/museumapplication/utils/Permission/PermissionInterface.java
package com.example.museumapplication.utils.Permission;

public interface PermissionInterface {
    /**
     * Get Permissions Request Code
     *
     * @return PermissionsRequestCode
     */
    int getPermissionsRequestCode();

    /**
     * Get Permissions
     *
     * @return Permissions
     */
    String[] getPermissions();

    /**
     * Request Permissions Success
     */
    void requestPermissionsSuccess();

    /**
     * Request Permissions Fail
     */
    void requestPermissionsFail();
}

File: app/src/main/java/com/example/museumapplication/utils/permission/PermissionInterface.kt
package com.example.museumapplication.utils.permission

interface PermissionInterface {
    /**
     * Get Permissions Request Code
     *
     * @return PermissionsRequestCode
     */
    fun permissionsRequestCode(): Int

    /**
     * Get Permissions
     *
     * @return Permissions
     */
    fun permissions(): Array<String?>?

    /**
     * Request Permissions Success
     */
    fun requestPermissionsSuccess()

    /**
     * Request Permissions Fail
     */
    fun requestPermissionsFail()
}